### PR TITLE
WorldControls entry and GameObject.Find uses

### PR DESCRIPTION
**Pull request by OlafSzmidt**

### CODE REVIEW

File: UnityProject/Assets/Scripts/Players/PlayerController.cs [hunk 1]
```diff
@@ -5,73 +5,44 @@
...
-    {
-        // General movement variables.
-        private const float speed = 2.0f;
-        private Vector3 velocity = new Vector3(0, 0, 0);
-        private const float moveInterval = 0.5f;
+	public class PlayerController : MonoBehaviour
+	{
+		// General movement variables.
+		private const float speed = 2.0f;
+		private Vector3 velocity = new Vector3 (0, 0, 0);
```
Comment (mrniket) [inline UnityProject/Assets/Scripts/Players/PlayerController.cs]: remove space here

File: UnityProject/Assets/Scripts/Players/PlayerManager.cs [hunk 2]
```diff
@@ -1,128 +1,137 @@
...
+		public static String PLAYER_TAG = "Avatar";
+		Dictionary<string, GameObject> activePlayers = new Dictionary<string, GameObject> ();
+
+		public bool CreatePlayer (PlayerDTO playerDTO)
+		{
+			// It might have already been created.
+			if (activePlayers.ContainsKey (PlayerId (playerDTO.id)))
+				return true;
+
+			GameObject player = GameObject.CreatePrimitive (PrimitiveType.Sphere);
```
Comment (mrniket) [inline UnityProject/Assets/Scripts/Players/PlayerManager.cs]: would prefer to use a prefab for this setup

Comment (mrniket) [reply UnityProject/Assets/Scripts/Players/PlayerManager.cs]: maybe in PlayerGenerator

File: UnityProject/Assets/Scripts/Players/PlayerManager.cs [hunk 3]
```diff
@@ -1,128 +1,137 @@
...
+
+	public class PlayerManager : MonoBehaviour, IPlayerManager
+	{
+		public static String PLAYER_TAG = "Avatar";
+		Dictionary<string, GameObject> activePlayers = new Dictionary<string, GameObject> ();
+
+		public bool CreatePlayer (PlayerDTO playerDTO)
+		{
+			// It might have already been created.
+			if (activePlayers.ContainsKey (PlayerId (playerDTO.id)))
```
Comment (mrniket) [inline UnityProject/Assets/Scripts/Players/PlayerManager.cs]: prefer to still put brackets here even if they are not needed

File: UnityProject/Assets/Scripts/Players/PlayerManager.cs [hunk 4]
```diff
@@ -1,128 +1,137 @@
...
+			if (player == null)
+				return false;
+
+			player.tag = PLAYER_TAG;
+			player.name = PlayerId (playerDTO.id);
+			player.AddComponent<PlayerController> ();
+
+			activePlayers.Add (PlayerId (playerDTO.id), player);
+
+			Debug.Log ("Added player");
```
Comment (mrniket) [inline UnityProject/Assets/Scripts/Players/PlayerManager.cs]: We don't need this debug log, we can see the results in the scene hierarchy :)

File: UnityProject/Assets/Scripts/Players/PlayerManager.cs [hunk 5]
```diff
@@ -1,128 +1,137 @@
...
+			Destroy (activePlayers [PlayerId (id)]);
+
+			activePlayers.Remove (PlayerId (id));
+
+			Debug.Log ("Deleted player");
+
+			return true;
+		}
+
+		public bool UpdatePlayer (PlayerDTO playerDTO)
```
Comment (mrniket) [inline UnityProject/Assets/Scripts/Players/PlayerManager.cs]: can we use the visual studio formatting rules here? (I think that means no space between `Player` and `(`

File: UnityProject/Assets/Scripts/Players/PlayerManager.cs [hunk 6]
```diff
@@ -1,128 +1,137 @@
...
+				return false;
+
+			PlayerController controller = playerToUpdate.GetComponent<PlayerController> ();
+			if (controller == null)
+				return false;
+
+			// The controller will change the position, score and health.
+			controller.SetNextState (playerDTO);
+
+			Debug.Log ("Updated player");
```
Comment (mrniket) [inline UnityProject/Assets/Scripts/Players/PlayerManager.cs]: we can remove this :)

File: UnityProject/Assets/Scripts/Players/PlayerManager.cs [hunk 7]
```diff
@@ -1,128 +1,137 @@
...
+
+			return true;
```
Comment (mrniket) [inline UnityProject/Assets/Scripts/Players/PlayerManager.cs]: do we do anything special from the method returns here?

File: UnityProject/Assets/Scripts/Players/PlayerManager.cs (hunk 3 above)
Comment (OlafSzmidt) [reply UnityProject/Assets/Scripts/Players/PlayerManager.cs]: I don't agree :(

File: UnityProject/Assets/Scripts/Players/PlayerManager.cs (hunk 2 above)
Comment (OlafSzmidt) [reply UnityProject/Assets/Scripts/Players/PlayerManager.cs]: Had some problems with this, will be done in the next PR

File: UnityProject/Assets/Scripts/Players/PlayerManager.cs (hunk 7 above)
Comment (OlafSzmidt) [reply UnityProject/Assets/Scripts/Players/PlayerManager.cs]: Not yet but it leaves us with possibility of doing something with it. Right now we check if it's null etc then return false but don't do anything on the other end with it, yet

**Pull request merged**